packages/react-codesandboxer/src/CodeSandboxer.js

```diff
@@ -192,20 +194,28 @@ export default class CodeSandboxDeployer extends Component<Props, State> {
   };
 
   componentDidUpdate(prevProps: Props) {
-    if (
-      this.props.examplePath !== prevProps.examplePath ||
-      this.props.name !== prevProps.name ||
-      this.props.gitInfo !== prevProps.gitInfo ||
-      this.props.example !== prevProps.example ||
-      this.props.pkgJSON !== prevProps.pkgJSON ||
-      this.props.importReplacements !== prevProps.importReplacements ||
-      this.props.dependencies !== prevProps.dependencies ||
-      this.props.providedFiles !== prevProps.providedFiles ||
-      this.props.extensions !== prevProps.extensions ||
-      this.props.template !== prevProps.template
-    ) {
+    /* If props related to loading files have been changed, next deploy should reload files */
+    /* The props that are compared should be the same as the arguments of fetchFiles */
+    const compareKeys = ['examplePath', 'gitInfo', 'importReplacements',
+      'dependencies', 'providedFiles', 'name', 'extensions', 'template']
+    if (!isEqual(pick(this.props, compareKeys), pick(prevProps, compareKeys))) {
       this.shouldReload = true;
+      return;
```

Comment:
Are we adding these returns because eslint says so?

Reply:
This return and the one on line 210 are there so that the code below it is not run (because shouldReload needs to be set anyway if anything is different).

The return on line 215 is largely redundant, but I put it there to allow more code to be written underneath in the same style. Happy to take that out though.

It is also possible to put the code below in an else statement, rather than using return. Which is better?

Reply:
I prefer putting them in else statements here, as it's odd to return from `shouldComponentUpdate`, even if you are only doing it to short-circuit the function further down the line.